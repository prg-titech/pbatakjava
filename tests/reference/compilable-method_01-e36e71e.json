{
    "basename": "compilable-method_01-e36e71e",
    "cmd": "{infile}",
    "infile": "tests/method_01/build.sh",
    "infile_hash": "b4ec96e3e1a1f172940e50aebe288067ea82840efd33f2fdb870ae55",
    "stdout": "compilable-method_01-e36e71e.stdout",
    "stdout_hash": "4b647133de82af23274e9e1513224939e1963af76f71369ea1563830",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}