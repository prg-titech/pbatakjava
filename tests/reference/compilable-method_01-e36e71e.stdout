
public interface Point {
  public int get__x();

  public int set__x(int toAssign);

  public int get__y();

  public int set__y(int toAssign);
}

public interface Point_Factory {
  public Point make(int x, int y);
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }
}

public class Point_v2 implements Point {
  public int x;

  public int y;

  public Point_v2(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v2(x, y);
  }
}

public interface Line {
  public Point get__p1();

  public Point set__p1(Point toAssign);

  public Point get__p2();

  public Point set__p2(Point toAssign);

  public Line update(Point_Factory Y, Point p);
}

public interface Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2);
}

public class Line_v1 implements Line {
  public Point_Factory V;

  public Point_Factory W;

  public Point p1;

  public Point p2;

  public Line_v1(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    this.V = V;
    this.W = W;
    this.p1 = p1;
    this.p2 = p2;
  }

  public Line update(Point_Factory Y, Point p) {
    return new Line_v1(V, Y, p1, p);
  }

  public Point get__p1() {
    return this.p1;
  }

  public Point set__p1(Point toAssign) {
    return this.p1 = toAssign;
  }

  public Point get__p2() {
    return this.p2;
  }

  public Point set__p2(Point toAssign) {
    return this.p2 = toAssign;
  }
}

public class Line_v1_Factory implements Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    return new Line_v1(V, W, p1, p2);
  }
}

public interface Test {
}

public interface Test_Factory {
}

public class Test_v1 implements Test {
  public static void main(String[] args) {
    Line l1 = new Line_v1(new Point_v1_Factory(), new Point_v2_Factory(), new Point_v1(1, 1), new Point_v2(2, 2));
    Line l2 = l1.update(new Point_v2_Factory(), new Point_v2(2, 2));
  }
}

public class Test_v1_Factory implements Test_Factory {
}
